doc/guides/using-symbols.md

```diff
@@ -0,0 +1,61 @@
+# Using global symbols
+
+ES6 introduced a new type: `Symbol`. This new type is _immutable_, and
+it is often used for metaprogramming purposes, as they can be used as
+property keys like string. There are two types of
+symbols, local and globals. Symbols are not included in
+JSON.stringify() but thay are by `util.inspect()`.
+
+## Local `Symbol`
+
+Local symbols are created with `Symbol(string)` and they are local to
+the caller function. Note that `Symbol('hello') !== Symbol('hello')`.
+For this reason, we often use them to simulate private fields, like so:
+
+```js
+const kField = Symbol('field');
+
+console.log(kField === Symbol('field')); // false
+
+class MyObject {
+  constructor() {
+    this[kField] = 'something';
+  }
+}
+
+module.exports.MyObject = MyObject;
+```
+
+Note that Symbols are not _fully private_, as the data could be accessed
+anyway:
+
+```js
+for (const s of Object.getOwnPropertySymbols(obj)) {
+  const desc = s.toString().replace(/Symbol\((.*)\)$/, '$1');
+  if (desc === 'field') {
+    console.log(obj[s]); // 'something'
+  }
+}
+```
+
+Local symbols make it harder for developers to monkey patch/access
+private fields, as they require more work than a property prefixed
+with an `_`.
```

Comment:
Say why this is good. If I were an unsuspecting developer I might consider

> make it harder for developers to monkey patch/access private fields

a bad thing and question why Node.js is doing this.